### PR TITLE
image: Add support for section TLV

### PR DESCRIPTION
**Pull request by agross-korg**
This patch adds support for the section TLV.  The Section TLV is a protected
TLV that supplies information about a specific elf section residing inside the
image.  There are some cases where only a image file is available and the
section information is required to do some post processing.  This TLV adds the
ability to deal with post processing in the absence of elf files.

Signed-off-by: Andy Gross <andy.gross@juul.com>

### CODE REVIEW

File: image/image.go [hunk 1]
```diff
@@ -67,6 +67,7 @@ const (
 	IMAGE_TLV_ENC_EC256 = 0x32
 	IMAGE_TLV_AES_NONCE = 0x50
 	IMAGE_TLV_SECRET_ID = 0x60
+	IMAGE_TLV_SECTION = 0x70
```
Comment (utzig) [inline image/image.go:70]: This should be aligned if `go fmt`  was used! :-P

File: image/image.go [hunk 2]
```diff
@@ -82,6 +83,7 @@ var imageTlvTypeNameMap = map[uint8]string{
 	IMAGE_TLV_ENC_EC256: "ENC_EC256",
 	IMAGE_TLV_AES_NONCE: "AES_NONCE",
 	IMAGE_TLV_SECRET_ID: "SEC_KEY_ID",
+	IMAGE_TLV_SECTION: "SECTION",
```
Comment (utzig) [inline image/image.go:86]: Need to align properly

File: image/image.go (hunk 1 above)
Comment (utzig) [reply image/image.go:70]: Btw, `0x50`, `0x60` and `0x70` are all downstream and not part of MCUboot. MCUboot already has its own `0x50` and `0x60` TLVs not related to these, so I just wonder when will the day come that you sync to MCUboot upstream and unexpected things happen! :-)

**Review by utzig — APPROVED** [review]
A few nits, that should be fixed, otherwise looks good.

File: image/image.go (hunk 1 above)
Comment (agross-korg) [reply image/image.go:70]: yeah good points on the TLV numbers.  What should we do for proprietary TLV numbers?  I wonder if we shouldnt set aside some proprietary range.

File: image/image.go [hunk 3]
```diff
@@ -55,33 +55,39 @@ const (
...
+	IMAGE_TLV_ECDSA224         = 0x21
+	IMAGE_TLV_ECDSA256         = 0x22
+	IMAGE_TLV_RSA3072          = 0x23
+	IMAGE_TLV_ED25519          = 0x24
+	IMAGE_TLV_ENC_RSA          = 0x30
+	IMAGE_TLV_ENC_KEK          = 0x31
+	IMAGE_TLV_ENC_EC256        = 0x32
+	IMAGE_TLV_AES_NONCE_LEGACY = 0x50
+	IMAGE_TLV_SECRET_ID_LEGACY = 0x60
+	IMAGE_TLV_AES_NONCE        = 0xa1
```
Comment (nkaje) [inline image/image.go:70]: Should we document why ranges 0x70 - 0xa0 not used? As it is used by MCUBoot?

Comment (agross-korg) [reply image/image.go:70]: The range is supposed to match mcuboot.

File: image/image.go (hunk 2 above)
Comment (agross-korg) [reply image/image.go]: done.

File: image/image.go (hunk 1 above)
Comment (agross-korg) [reply image/image.go]: moved these to legacy numbers.  we'll keep them around until we update mcuboot locally and then remove them then.

File: image/image.go (hunk 3 above)
Comment (agross-korg) [reply image/image.go:70]: mcuboot drives the fixed TLV numbers.

**Pull request merged**